src/module-loader/dotnet/src/MorganStanley.ComposeUI.ModuleLoader.Abstractions/NativeStartupProperties.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+namespace MorganStanley.ComposeUI.ModuleLoader;
+
+/// <summary>
+/// Represents startup properties specific to native modules.
+/// </summary>
+public sealed class NativeStartupProperties : ModuleStartupProperties
+{
+    /// <summary>
+    /// Gets or sets the collection of environment variables to be set for the native module at startup.
+    /// </summary>
+    public Dictionary<string, string> EnvironmentVariables { get; } = new();
```

Comment:
The original EnvironmentVariables class is deliberately created like that to prevent different modules manipilating each other's variables.